File: omnigate/ts/gate.ts
#!/usr/bin/env -S deno run -A
/** Deno: merge-gate with cwd set on the command (no global chdir). */
const root = Deno.env.get("CREATION_OS_ROOT") ?? ".";
const c = new Deno.Command("make", {
  args: ["merge-gate"],
  cwd: root,
  stdout: "inherit",
  stderr: "inherit",
});
const { code } = await c.output();
Deno.exit(code);

File: omnigate/julia/gate.jl
#!/usr/bin/env julia
# Julia: run merge-gate from CREATION_OS_ROOT
cd(get(ENV, "CREATION_OS_ROOT", "."))
try
    run(`make merge-gate`)
catch e
    if e isa Base.ProcessFailedException
        exit(e.procs[end].exitcode)
    end
    rethrow(e)
end
